src/ports/linux/mib/system_mib.c

```diff
@@ -0,0 +1,575 @@
+    * SET REQUEST
+    *
+    * multiple states in the transaction.  See:
+    * http://www.net-snmp.org/tutorial-5/toolkit/mib_module/set-actions.jpg
+    */
+   case MODE_SET_RESERVE1:
+      ret = netsnmp_check_vb_type_and_max_size (
+         requests->requestvb,
+         ASN_OCTET_STR,
+         sizeof (sysname.string) - 1);
+      if (ret != SNMP_ERR_NOERROR)
+      {
+         netsnmp_set_request_error (reqinfo, requests, ret);
+      }
+      break;
+
+   case MODE_SET_RESERVE2:
+      old_sysname = SNMP_MALLOC_TYPEDEF (pf_snmp_system_name_t);
+      if (old_sysname == NULL)
+      {
+         netsnmp_set_request_error (
+            reqinfo,
+            requests,
+            SNMP_ERR_RESOURCEUNAVAILABLE);
+      }
+      pf_snmp_get_system_name (pnet, old_sysname);
+      break;
+
+   case MODE_SET_FREE:
+      if (old_sysname != NULL)
+      {
+         SNMP_FREE (old_sysname);
+      }
+      break;
+
+   case MODE_SET_ACTION:
+      memcpy (
+         sysname.string,
+         requests->requestvb->val.string,
+         requests->requestvb->val_len);
+      sysname.string[requests->requestvb->val_len] = 0;
+      ret = pf_snmp_set_system_name (pnet, &sysname);
+      if (ret)
+      {
+         netsnmp_set_request_error (reqinfo, requests, SNMP_ERR_GENERR);
+      }
+      break;
+
+   case MODE_SET_COMMIT:
+      SNMP_FREE (old_sysname);
+      break;
+
+   case MODE_SET_UNDO:
+      ret = pf_snmp_set_system_name (pnet, old_sysname);
+      if (ret)
+      {
+         netsnmp_set_request_error (reqinfo, requests, SNMP_ERR_UNDOFAILED);
+      }
+      SNMP_FREE (old_sysname);
+      break;
+
+   default:
+      /* we should never get here, so this is a really bad error */
+      LOG_ERROR (
+         PF_SNMP_LOG,
+         "system_mib(%d): unknown mode (%d)\n",
+         __LINE__,
+         reqinfo->mode);
+      return SNMP_ERR_GENERR;
+   }
+
+   return SNMP_ERR_NOERROR;
+}
+
+int handle_sysLocation (
+   netsnmp_mib_handler * handler,
+   netsnmp_handler_registration * reginfo,
+   netsnmp_agent_request_info * reqinfo,
+   netsnmp_request_info * requests)
+{
+   int ret;
+   pnet_t * pnet = reginfo->my_reg_void;
+   pf_snmp_system_location_t syslocation;
+   static pf_snmp_system_location_t * old_syslocation = NULL;
+
+   /* We are never called for a GETNEXT if it's registered as a
+      "instance", as it's "magically" handled for us.  */
+
+   /* a instance handler also only hands us one request at a time, so
+      we don't need to loop over a list of requests; we'll only get one. */
+
+   switch (reqinfo->mode)
+   {
+
+   case MODE_GET:
+      pf_snmp_get_system_location (pnet, &syslocation);
+      snmp_set_var_typed_value (
+         requests->requestvb,
+         ASN_OCTET_STR,
+         syslocation.string,
+         strlen (syslocation.string));
+      break;
+
+   /*
+    * SET REQUEST
+    *
+    * multiple states in the transaction.  See:
+    * http://www.net-snmp.org/tutorial-5/toolkit/mib_module/set-actions.jpg
```

Comment:
```suggestion
    * http://www.net-snmp.org/tutorial/tutorial-5/toolkit/mib_module/set-actions.jpg
```

Reply:
The link is generated by the net-snmp provided mib2c code. It doesn't currently work but it's not clear to me if this is a temporary problem or if it has moved permanently. I'll fix it for now and hopefully it will continue to be available at the new location.